### PR TITLE
nixosTests.bees: port to python

### PR DESCRIPTION
**Pull request by flokli**
###### Motivation for this change
#72828

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [x] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @

### CODE REVIEW

File: nixos/tests/bees.nix [hunk 1]
```diff
@@ -29,27 +29,31 @@ import ./make-test.nix ({ lib, ... }:
 
   testScript =
   let
-    withRetry = content: maxTests: sleepTime: ''
-      max_tests=${lib.escapeShellArg maxTests}; sleep_time=${lib.escapeShellArg sleepTime}; for ((i=0; i<max_tests; i++)); do ${content} && exit 0; sleep "$sleep_time"; done; exit 1;
+    someContentIsShared = loc: pkgs.writeScript "some-content-is-shared" ''
```
Comment (worldofpeace) [inline nixos/tests/bees.nix]: Can we use `writeShellScript`? It puts the correct shebang right away `runtimeShell`.

**Review by worldofpeace — CHANGES_REQUESTED** [review]

**Comment by worldofpeace** [comment]
We also need to add `bees` to `all-tests.nix`.

**Comment by worldofpeace** [comment]
@GrahamcOfBorg test bees

File: nixos/tests/bees.nix [hunk 2]
```diff
@@ -29,27 +29,31 @@ import ./make-test.nix ({ lib, ... }:
...
-    $machine->succeed("cp --reflink=never /aux1/dedup-me-1 /aux1/dedup-me-2");
-    $machine->succeed("cp --reflink=never /aux1/* /aux2/");
-    $machine->succeed("sync");
-    $machine->fail(q(${someContentIsShared "/aux1"}));
-    $machine->fail(q(${someContentIsShared "/aux2"}));
-    $machine->succeed("systemctl start beesd\@aux1.service");
+    machine.succeed("dd if=/dev/urandom of=/aux1/dedup-me-1 bs=1M count=8")
+    machine.succeed("cp --reflink=never /aux1/dedup-me-1 /aux1/dedup-me-2")
+    machine.succeed("cp --reflink=never /aux1/* /aux2/")
+    machine.succeed("sync")
```
Comment (tfc) [inline nixos/tests/bees.nix]: these are a lot of commands on the same machine without use of intermediate outputs. how about combining them all in one `succeed` call to increase readability?

Comment (flokli) [reply nixos/tests/bees.nix]: Updated, thanks.

File: nixos/tests/bees.nix (hunk 1 above)
Comment (flokli) [reply nixos/tests/bees.nix]: Done.

**Comment by flokli** [comment]
@worldofpeace I adressed all changes, except adding the test to `passthru` - `bees` is packaged in a slightly weird way, didn't really want to touch that.

**Pull request merged**